### PR TITLE
Move map filter to the root level and add mutually exclusive filters in configuration

### PR DESCRIPTION
**Pull request by vadymmarkov**
# Why?

In the FINN app "location" and "map" filters are mutually exclusive and it doesn't make sense to select them together. At the same time it's not the best UX to clear out "location" when "map" is selected and the other way around.

# What?

- We decided to move "map" filter to the root level and disable "location" filter  when "map" is selected. It works both ways and not only for those two filters. It's possible to specify your custom mutually exclusive filters in `FilterConfiguration` now.

-  Show radius as a tag to indicate that map filter is selected

- Rename string properties from FilterConfiguration to not confuse the type system

# Show me

simulator screen shot - iphone xs - 2019-02-20 at 15 49 32

### CODE REVIEW

File: Sources/Core/Filters/Root/RootFilterViewController.swift [hunk 1]
```diff
@@ -39,7 +39,12 @@ final class RootFilterViewController: FilterViewController {
     }()
 
     private var searchFilter: Filter? {
-        return filter.subfilters.first { $0.key == config.searchFilterKey }
+        return filter.subfilters.first { $0.key == config.searchFilter }
+    }
+
+    private func predicateForMutuallyExclusiveFilters(excluding filter: Filter) -> ((Filter) -> Bool) {
```
Comment (shredlocker) [inline Sources/Core/Filters/Root/RootFilterViewController.swift]: This function isn't used anywhere

File: Sources/FINN/BackendModel/FilterSetup.swift [hunk 2]
```diff
@@ -50,33 +50,35 @@ public struct FilterSetup: Decodable {
...
-                locationKey: "geoLocationName"
+                key: key,
+                latitudeKey: FilterKey.latitude.rawValue,
+                longitudeKey: FilterKey.longitude.rawValue,
+                radiusKey: FilterKey.radius.rawValue,
+                locationKey: FilterKey.geoLocationName.rawValue
             )
-            locationFilter.add(subfilter: mapFilter, at: 0)
+
+            let index = supportedFilters.firstIndex(where: { $0.key == config.locationFilter }) ?? 0
```
Comment (shredlocker) [inline Sources/FINN/BackendModel/FilterSetup.swift]: Could add map key into the `supportedFilters` at the correct index in the config.

File: Sources/Core/Filters/Root/RootFilterViewController.swift (hunk 1 above)
Comment (vadymmarkov) [reply Sources/Core/Filters/Root/RootFilterViewController.swift]: Nice catch, it's a leftover from my experiments 😅

File: Sources/FINN/BackendModel/FilterSetup.swift (hunk 2 above)
Comment (vadymmarkov) [reply Sources/FINN/BackendModel/FilterSetup.swift]: Great idea, now I know what's `supportedFilters` for 😄

Comment (shredlocker) [reply Sources/FINN/BackendModel/FilterSetup.swift]: We could maybe come up with a better name for those filters :P

Comment (shredlocker) [reply Sources/FINN/BackendModel/FilterSetup.swift]: `RootLevelFilters` or something is a bit more descriptive

Comment (vadymmarkov) [reply Sources/FINN/BackendModel/FilterSetup.swift]: @shredlocker Should be fixed now 😉

**Pull request merged**